Refactor: all dialogs now use <Dialog> headlessui component (and more).

hansl opened the pull request:
This PR took a turn and now I have fixed all the dialogs that I could find. This removes a lot of code and `useEffect` in favor of simpler state management and dialog.

This makes it more streamline when we have a stack of dialogs. Showing multiple dialogs at once and pressing Escape will close the latest opened one. Same with closing dialogs using buttons or actions. This change should have no visible change (outside of going up the stack once when pressing Escape or clicking outside the modal).

The pattern is more or less the same through the app: get a current dialog, either move it to its own component or refactor it into a headlessui `<Dialog>`, and reduce the usage of styles and backdrop to simpler classes.

Also, moved the `SignModal` to a `Web3Auth` context and provider, and create a new component where anyone who needs to `useTx` also include the `<SignModal>` in the right DOM location to show it in their stack.

Without actual e2e tests this is hard to test. I fixed the unit tests we already had. I did manually test all dialogs as I went.

If you find a bug that relates to dialogs or UX flow related to signing, let me know.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **UI Improvements**
	- Redesigned modal dialogs now offer enhanced accessibility, consistent styling, and a more intuitive interface.
	- Added a backdrop to modals for improved visual presentation.
- **Enhanced Transaction Experience**
	- Integrated an improved signing interface that delivers clear feedback during transaction approvals.
- **Dynamic Validation**
	- Voting and group policy forms now present real-time, human-readable guidance based on configurable minimum durations.
- **New Features**
	- Introduced `SignModal` components across various modals for signing transactions.
- **Bug Fixes**
	- Improved modal management and state handling, enhancing overall user experience.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

chalabi2 reviewed (commented):
Thank you for this!

Everything looks good and functions well. 

Couple minor issues:

- backdrop on send modal isnt working
- I think all modals need to be pushed right by the width of the sideNav in its extended state. This is especially visible on the mint and burn modals in the tokenfactory denom list. 
- @fmorency do we want animations for dialog open and close.

hansl commented:
Sorry I'm going to put this in draft again, it is not ready.

fmorency reviewed (changes requested):
Thanks! 

I glanced at the code and saw nothing major with my limited FE skills.

Can you fix the build, please?

fmorency commented on components/admins/modals/descriptionModal.tsx:
```diff
@@ -1,33 +1,55 @@
+
+      <Dialog.Panel>
+        <form method="dialog" className="modal-box ">
+          <button
+            className="btn btn-sm btn-circle btn-ghost absolute right-2 top-2"
+            aria-label="x-close"
+          >
+            ✕
+          </button>
```
@hansl Relevant?

hansl commented on bun.lock:
Don't know, re-added.

hansl commented on components/groups/modals/index.tsx:
```diff
@@ -1,5 +1,5 @@
 export * from '../../react/authSignerModal';
-export * from './groupDetailsModal';
+//export * from './groupDetailsModal'; // Do we need this?
```
Yeah I thought about it a bit more and if we need this component we can always take it back from git. Removed

hansl commented on config/env.ts:
```diff
@@ -1,4 +1,6 @@
 const env = {
+  production: process.env.NODE_ENV === 'production',
```
It's used in the manifest app provider for the logging of `ChainProvider`. My goal is going to be using it for more stuff, e.g. extra logging.

hansl commented on components/react/authSignerModal.tsx:
```diff
@@ -238,39 +242,72 @@
+      <Dialog.Panel className="modal-box max-w-lg w-full dark:bg-[#1D192D] bg-[#FFFFFF] rounded-lg shadow-xl">
         <div className="flex justify-between items-center pb-4">
           <div className="flex items-center gap-3">
             <img
```
I'll address lint errors in another PR. There are a bunch of these that I want to see if they're a good idea, but it's out of scope for an already long PR that's focusing on dialogs.

fmorency reviewed (changes requested):
Thanks! A few comments

- Pressing the `x` button in the sign modal does nothing
  ![2025-02-12_13-28](https://github.com/user-attachments/assets/b483ada5-c68d-4c97-a4c8-f990a700ec21)
- Pressing `Execute` on a proposal gives the following error
  ![2025-02-12_13-34](https://github.com/user-attachments/assets/47f6990c-5828-4ace-a75e-ac79220901d6)

- Withdrawing a proposal and rejecting the transaction closes all modals
- Voting on a proposal and clicking the `x` button on the "Case your vote" dialog closes the dialog and returns to the proposal. Clicking on the `Vote` button again shows the sign modal.
  ![2025-02-12_13-46](https://github.com/user-attachments/assets/8766dc9d-95f4-47da-a29a-f83c158508fe)
  ![2025-02-12_13-47](https://github.com/user-attachments/assets/ca1f8411-59ae-44e0-89ff-ba3b9565e0d1)

Might not be related to this PR?:
- I couldn't see my groups after login. I had to refresh + re-login. I'm using `Email` as my login.
  
  No group?
  ![2025-02-12_13-36](https://github.com/user-attachments/assets/d30dbbad-3323-40c8-a4d6-29c5a9881de3)
  
  After re-login
  ![2025-02-12_13-37](https://github.com/user-attachments/assets/39397b28-9092-40dc-b055-69defc10070c)
  
  I also noticed this behavior after creating a new token. I had to refresh + re-login to see it.

hansl commented:
@fmorency PTAL.

> Pressing the x button in the sign modal does nothing
Fixed

> Pressing Execute on a proposal gives the following error
Didn't get that error but didn't get a signing modal either. Fixed that.

> Withdrawing a proposal and rejecting the transaction closes all modals

Should be fixed now.

> Voting on a proposal and clicking the x button on the "Case your vote" dialog closes the dialog and returns to the proposal. Clicking on the Vote button again shows the sign modal.

Re-did the flow here; now the popup closes and the signing happens outside of it. It fixes a few bugs I had with that dialog. The right way to fix would be in general have a Flow-like state management, but that's a refactor I'm not willing to do. This is fine.

Fixed.

> I couldn't see my groups after login. I had to refresh + re-login. I'm using Email as my login.
This might be fixed by my last commit which calls refetch more often on voting. Maybe not.

fmorency reviewed (approved):
Thank you! ACK!

The following behavior is still present but we can fix on another PR.

- Withdrawing a proposal and rejecting the transaction closes all modals
